src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,181 @@
+        public virtual IHtmlLocalizer WithCulture([NotNull] CultureInfo culture) =>
+            new HtmlLocalizer(_localizer.WithCulture(culture), _encoder);
+
+        /// <summary>
+        /// Creates a new <see cref="IStringLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        IStringLocalizer IStringLocalizer.WithCulture([NotNull] CultureInfo culture) => WithCulture(culture);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public virtual IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments)
+        {
+            var stringValue = _localizer[key].Value;
+
+            return ToHtmlString(new LocalizedString(key, EncodeArguments(stringValue, arguments)));
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="LocalizedHtmlString"/> for a <see cref="LocalizedString"/>.
+        /// </summary>
+        /// <param name="result">The <see cref="LocalizedString"/>.</param>
+        protected virtual LocalizedHtmlString ToHtmlString(LocalizedString result) =>
+            new LocalizedHtmlString(result.Name, result.Value, result.ResourceNotFound);
+
+        /// <summary>
+        /// Encodes the arguments based on the object type.
+        /// </summary>
+        /// <param name="resourceString">The resourceString whose arguments need to be encoded.</param>
+        /// <param name="arguments">The array of objects to encode.</param>
+        /// <returns>The string with encoded arguments.</returns>
+        protected virtual string EncodeArguments([NotNull] string resourceString, [NotNull] object[] arguments)
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            char currentCharacter;
+            StringBuilder tokenBuffer = null;
+            var outputBuffer = new StringBuilder();
+            var isToken = false;
+
+            while (position < length)
+            {
+                currentCharacter = resourceString[position];
+
+                position++;
+                if (currentCharacter == '}')
+                {
+                    tokenBuffer.Append(currentCharacter);
```

Comment:
will NRE if `tokenBuffer` is `null`. Should be valid for resource to start with "}}"